### PR TITLE
Mash assign worker to referral

### PR DESCRIPTION
**Pull request by moarpheus**
## Link to JIRA ticket

[JIRA ticket](https://hackney.atlassian.net/browse/SCT-1570)

### *What is the problem we're trying to solve*

Previously API for MASH referrals didn't include functionality to assign Worker to the Referral. The proposed code changes allows that.

### *What changes have we introduced*

Rules in the [boundary](https://github.com/LBHackney-IT/social-care-case-viewer-api/blob/mash-assign-worker-to-referral/SocialCareCaseViewerApi/V1/Boundary/Requests/UpdateMashReferral.cs) to enforce request to allow only for Worker email or Id must be supplied and of correct types with [tests](https://github.com/LBHackney-IT/social-care-case-viewer-api/blob/mash-assign-worker-to-referral/SocialCareCaseViewerApi.Tests/V1/Boundary/Request/UpdateMashReferralTests.cs). 

Integration [test](https://github.com/LBHackney-IT/social-care-case-viewer-api/blob/mash-assign-worker-to-referral/SocialCareCaseViewerApi.Tests/V1/IntegrationTests/MASH/AssignWorkerToMashReferralTests.cs) with helpers.

Factory additions, minor changes in controller, mash referral, gateway to support functionality and minor refactoring + dotnet formatting.

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

n/a

### CODE REVIEW

File: SocialCareCaseViewerApi.Tests/V1/Controllers/MashReferral/UpdateMashReferralTests.cs [hunk 1]
```diff
@@ -33,9 +33,10 @@ public void ValidationReturnsBadRequestIfUpdateTypeIsContactDecisionAndNoUrgentC
         {
             var request = TestHelpers.CreateUpdateMashReferral(updateType: "CONTACT-DECISION");
             request.RequiresUrgentContact = null;
+            request.WorkerId = null;
 
             var response = _mashReferralController.UpdateMashReferral(request, _fakeReferralId) as ObjectResult;
-
+            //set email or id to null
```
Comment (Miles-Alford) [inline SocialCareCaseViewerApi.Tests/V1/Controllers/MashReferral/UpdateMashReferralTests.cs:39]: Do we need this comment anymore? it looks like the id is set to null on line 36

File: SocialCareCaseViewerApi.Tests/V1/IntegrationTests/MASH/AssignWorkerToMashReferralTests.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        [SetUp]
+        public void Setup()
+        {
+            // Clear test database of any rows in the database
+            DatabaseContext.Teams.RemoveRange(DatabaseContext.Teams);
+            DatabaseContext.Workers.RemoveRange(DatabaseContext.Workers);
+            DatabaseContext.MashReferrals.RemoveRange(DatabaseContext.MashReferrals);
+
+            // Create existing refferal and unrelated worker
```
Comment (Miles-Alford) [inline SocialCareCaseViewerApi.Tests/V1/IntegrationTests/MASH/AssignWorkerToMashReferralTests.cs:26]: tiny typo in the comment

File: SocialCareCaseViewerApi.Tests/V1/IntegrationTests/MASH/AssignWorkerToMashReferralTests.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            var (existingDbWorker, _) = IntegrationTestHelpers.SetupExistingWorker(DatabaseContext);
+            var existingDbRefferal = IntegrationTestHelpers.SaveMashReferralToDatabase(DatabaseContext);
```
Comment (Miles-Alford) [inline SocialCareCaseViewerApi.Tests/V1/IntegrationTests/MASH/AssignWorkerToMashReferralTests.cs:28]: typo in variable name

File: SocialCareCaseViewerApi/V1/Controllers/CaseStatusController.cs [hunk 4]
```diff
@@ -120,7 +120,6 @@ public IActionResult CreateCaseStatusAnswers(CreateCaseStatusAnswerRequest reque
         /// Edit a Case Status
```
Comment (martapederiva) [inline SocialCareCaseViewerApi/V1/Controllers/CaseStatusController.cs:120]: CaseStatus file

File: SocialCareCaseViewerApi/V1/Controllers/CaseStatusController.cs [hunk 5]
```diff
@@ -120,7 +120,6 @@ public IActionResult CreateCaseStatusAnswers(CreateCaseStatusAnswerRequest reque
...
         /// </summary>
         /// <param name="request"></param>
```
Comment (martapederiva) [inline SocialCareCaseViewerApi/V1/Controllers/CaseStatusController.cs:122]: CaseStatus file

File: SocialCareCaseViewerApi/V1/Domain/CaseStatus.cs [hunk 6]
```diff
@@ -8,11 +8,11 @@ namespace SocialCareCaseViewerApi.V1.Domain
     public class CaseStatus
     {
```
Comment (martapederiva) [inline SocialCareCaseViewerApi/V1/Domain/CaseStatus.cs:9]: CaseStatus

File: SocialCareCaseViewerApi/V1/Infrastructure/CaseStatusAnswer.cs [hunk 7]
```diff
@@ -26,13 +26,13 @@ public class CaseStatusAnswer : IAuditEntity
         public long CaseStatusId { get; set; }
```
Comment (martapederiva) [inline SocialCareCaseViewerApi/V1/Infrastructure/CaseStatusAnswer.cs:26]: CaseStatus

**Review by Miles-Alford — APPROVED** [review]
Looks good, good job!

**Pull request merged**